support-frontend/assets/components/subscriptionCheckouts/address/postcodeFinder.tsx

```diff
@@ -9,123 +8,148 @@ import {
-				data-qm-masking="blocklist"
-				onChange={(e) => {
-					setPostcode(e.target.value);
-					onPostcodeUpdate(e.target.value);
-				}}
-				isLoading={isLoading}
-				value={postcode ?? ''}
-			/>
-			{results.length > 0 && (
-				<Select
-					onChange={(e: React.ChangeEvent<HTMLSelectElement>) => {
-						const resultIndex = Number.parseInt(e.currentTarget.value);
-						if (results[resultIndex]) {
-							onAddressUpdate(results[resultIndex]);
-						}
-					}}
-					id="address"
-					label={`${results.length} addresses found`}
-				>
-					<Option value={''}>Select an address</Option>
-					<>
-						{results.map((result, key) => (
-							<Option key={result.lineOne} value={key}>
-								{[result.lineOne, result.lineTwo].join(', ')}
-							</Option>
-						))}
-					</>
-				</Select>
+		<div css={styles.container}>
+			<div css={styles.inputAndButtonContainer}>
+				<div>
+					<TextInput
+						label="Postcode"
+						value={postcode}
+						error={error}
+						onChange={(e) => {
+							onPostcodeUpdate(e.target.value);
+						}}
+						onKeyPress={(e) => {
+							if (e.key && e.key === 'Enter') {
+								e.preventDefault();
+								handleSubmit();
+							}
+						}}
+						data-qm-masking="blocklist"
+						width={10}
+					/>
+				</div>
+
+				{!isLoading && (
+					<div css={styles.buttonContainer}>
+						<ThemeProvider theme={buttonThemeReaderRevenueBrandAlt}>
+							<Button priority="tertiary" type="button" onClick={handleSubmit}>
+								Find address
+							</Button>
+						</ThemeProvider>
+					</div>
+				)}
+			</div>
+
+			{results && results.length > 0 && (
+				<div css={styles.addressSelectorContainer}>
+					<AddressSelector
+						results={results}
+						onAddressSelected={onAddressSelected}
+					/>
+				</div>
 			)}
 		</div>
 	);
 }
+
+// ---- Helper component ---- //
+
+interface AddressSelectorProps {
+	results: PostcodeFinderResult[];
+	onAddressSelected: (address: PostcodeFinderResult) => void;
+}
+
+function AddressSelector({ results, onAddressSelected }: AddressSelectorProps) {
+	function handleAddressSelected(e: React.ChangeEvent<HTMLSelectElement>) {
+		const resultIndex = Number.parseInt(e.currentTarget.value);
+		if (results[resultIndex]) {
+			onAddressSelected(results[resultIndex]);
+		}
+	}
+
+	return (
+		<Select
+			onChange={handleAddressSelected}
+			label={`${results.length} ${pluralisedAddress(results.length)} found`}
+		>
+			<Option value={''}>Select an address</Option>
+			<>
+				{results.map((result, index) => (
+					<Option key={result.lineOne} value={index}>
+						{addressOption(result)}
+					</Option>
+				))}
+			</>
+		</Select>
+	);
+}
+
+// ---- Helper functions ---- //
+
+function pluralisedAddress(numAddresses: number) {
+	return numAddresses === 1 ? 'address' : 'addresses';
+}
+
+function addressOption(address: PostcodeFinderResult) {
+	if (address.lineTwo) {
+		return [address.lineOne, address.lineTwo].join(', ');
+	}
+	return address.lineOne ?? '';
+}
```

Comment:
Previously, we just did `[address.lineOne, address.lineTwo].join(', ')` which means if we only have a `address.lineOne` we have a trailing comma which looks a little weird. 

We still have a slight issue in that if we don't have a `lineOne` we'll just show an empty string (although before we'd just show `, `, so maybe it's still better than that?)

Perhaps we should filter them out of the results? We can do this as a follow up